### DISCUSSION TITLE
Connectors and Dragging Nodes Should Be Improved

### DISCUSSION THREAD

**Discussion by autoca** [freeplane/freeplane#668] [Ideas]
Hello all,
Hello Dimitry,

First off, sorry for not replying a lot in these days to all those good threads that concern me.
I have not as much time at the moment. Learning new things at my new job and taking care of initial organizational things take away too much time and energy. On top of that, I am sick now and continue working in home office. But anyway, I have the following issues that continue to bother me enough that I still want to report them.

I continuously happen to click on a connector when I want to move the map.
This is extremely irritating, because then the connector gets pulled instead of the map moved.

1) Suggestion: I would wish that connectors can only be pulled with the mouse when I click control at the same time.

2) While we are at it: The same could be done for dragging nodes with the mouse (on top of other nodes, thus, changing the structure).

3) I also want to suggest to replace that weired blue too thin oval handle (for moving a node without changing the structure) with a mouse key + some keyboard modifier as well - or make the blue oval wider.

4) For Connectors, I also have other suggestions as well: They should be easier to modify and manage. Maybe I don't understand the concept of pulling their end sections but as it is now, it does not work for me. Also, I wish that I could select connectors as I can select nodes and do all kinds of stuff with a group of them at once, e.g. assign a new style to them.

MTIA and regards, a.

**Comment by euu2021** [comment]
Part of those suggestions were discussed [here](https://sourceforge.net/p/freeplane/discussion/758437/thread/d230065b5c/). About using keyboard modifiers, Dimitry explained:

> I do not think that overriding mouse behavior using keyboard hot keys makes anything easier. Hot keys are already heavily used for drag and drop operations to switch between moving, copying and creating connectors and adding complexity here can only increase confusion.

---
I'm curious about your problem with the oval handle. Do you think they are too difficult to hover? 

---
About unintentional interaction while trying to move the map,[ this suggestion](https://github.com/freeplane/freeplane/discussions/388) is also a possible solution.

---
> Maybe I don't understand the concept of pulling their end sections but as it is now, it does not work for me.

There is a guide [here](https://docs.freeplane.org/attic/new-wiki-initiative/understanding-nodes.html?highlight=handle#connectors).

---
>Also, I wish that I could select connectors as I can select nodes and do all kinds of stuff with a group of them at once, e.g. assign a new style to them.

I don't understand about the coding part of FP, but this feature sounds technically difficult to implement (I may be wrong, though). So, if I was really interested in that, I would try to achieve something similar using scripts.

Connectors is an area of FP with a lot of unexplored potential, so a lot can be done (and was done) with scripts. Allan created [this](https://www.youtube.com/watch?v=cSyhkcsB2-M&t=1501s) impressive add-on, but it's not maintained, so it's buggy, now. There are other scripts along the forums.

So, in order to create a connector style manager, I would start by trying this:

- use [this](https://docs.freeplane.org/scripting/Scripts_collection.html#template-chooser---create-nodes-from-templates) to have a GUI with a combobox
- read the comments by Edo [here](https://github.com/freeplane/freeplane/discussions/415) to seek a way to populate the combobox list with the connector styles
- read the scripts mentioned [here](https://sourceforge.net/p/freeplane/discussion/758437/thread/00a59ea0d6/), to see how to make simple references to groups of connectors. After that, studying the Mapinsight addon mentioned above will show a lot of other ways.

**Reply by autoca** [reply]
Hey euu, thx for the feedback and suggestions,

> Part of those suggestions were discussed [here](https://sourceforge.net/p/freeplane/discussion/758437/thread/d230065b5c/). About using keyboard modifiers, Dimitry explained:
> 
> > I do not think that overriding mouse behavior using keyboard hot keys makes anything easier. Hot keys are already heavily used for drag and drop operations to switch between moving, copying and creating connectors and adding complexity here can only increase confusion.

Where are hotkeys already used for connectors in this way? I do not notice that in my installation.

> I'm curious about your problem with the oval handle. Do you think they are too difficult to hover?

Yes, they are way too thin, especially on zoomed out maps for a better overview when the nodes are relatively small.
The oval handle shrinks in size with the zoom-out-factor, just like the nodes.
On the other hand, it also happens sometimes that I accidently move a node with that oval handle and then I have to find it again and double click it to set it's position back to default. Cumbersome.
 
> About unintentional interaction while trying to move the map,[ this suggestion](https://github.com/freeplane/freeplane/discussions/388) is also a possible solution.

Absolutely, but it is not implemented, or did I overlook any settings?

> > Maybe I don't understand the concept of pulling their end sections but as it is now, it does not work for me.
> 
> There is a guide [here](https://docs.freeplane.org/attic/new-wiki-initiative/understanding-nodes.html?highlight=handle#connectors).

Sure, but even the guide admits: "You'll often need to repeat your click-drag-release several times before you get the expected result, i.e. without affecting the far end." And that really is the worst part. Not useable like that. Horrrrrrrrrrrrrrrrrrrrrrrrrrrrrrible.

> > Also, I wish that I could select connectors as I can select nodes and do all kinds of stuff with a group of them at once, e.g. assign a new style to them.
> 
> I don't understand about the coding part of FP, but this feature sounds technically difficult to implement (I may be wrong, though). So, if I was really interested in that, I would try to achieve something similar using scripts.

Well other mindmap software can do it. I don't know what should be difficult about that unless the existing code base is messed up somehow. It can be done with nodes. Why not with the connectors as well? As of now, it is almost unusable.

Regards, a.

**Reply by dpolivaev** [reply]
I increased the width of the fragging handle and even made it configurable
Preview 1.10.5_03

**Reply by autoca** [reply]
Thank you!

**Comment by Illegit** [comment]
Useful. How to configure it?

**Reply by autoca** [reply]
It seems like it is not even published yet, just announced for next beta.

**Reply by macmarrum** [reply]
Node-drag handle is already available and published as a preview `Freeplane-1.10.5-pre03`. You can find it at https://sourceforge.net/projects/freeplane/files/freeplane%20preview/

To find the new option in preferences, I used Command Search (Ctrl+F1) and looked for "dragging".

**Reply by autoca** [reply]
Ah, ok, I was looking in the subfolder 1.10.x, oh well...

**Reply by Illegit** [reply]
Hi macmarrum: thankyou.
